### ISSUE TITLE
Fragments and `Can't find field <field> on result object ROOT_QUERY`

### ISSUE THREAD

**Issue by kamilkisiela** [apollostack/apollo-client#451]
Pattern:

https://gist.github.com/kamilkisiela/43964606ee4d0ee436257be719b93b0a

Repro:

https://github.com/apollostack/GitHunt-angular2/tree/issue-fragments

An error (from different project so don't look at paths):

```
store.ts:63
    Error: Can't find field <some field> on result object ROOT_QUERY.
    at eval (http://localhost:4200/vendor/apollo-client/data/writeToStore.js:56:23)
    at Array.forEach (native)
    at Object.writeSelectionSetToStore (http://localhost:4200/vendor/apollo-client/data/writeToStore.js:50:29)
    at Object.data (http://localhost:4200/vendor/apollo-client/data/store.js:29:43)
    at apolloReducer (http://localhost:4200/vendor/apollo-client/store.js:24:27)
    at eval (http://localhost:4200/vendor/redux/dist/redux.min.js:1:6058)
    at s (http://localhost:4200/vendor/redux/dist/redux.min.js:1:1956)
    at Object.eval [as dispatch] (http://localhost:4200/vendor/apollo-client/store.js:10:16)
    at eval (http://localhost:4200/vendor/apollo-client/QueryManager.js:412:33)
    at ZoneDelegate.invoke (http://localhost:4200/vendor/zone.js/dist/zone.js:323:29)
```

**Comment by Poincare** [comment]
@kamilkisiela Is this on the latest update of Apollo Client? We made some pretty significant changes to the store format so this may have been an unintended consequence.

**Comment by kamilkisiela** [comment]
Latest at the time I checked this, it was 7h ago.

pt., 22 lip 2016, 19:17 użytkownik Dhaivat Pandya <notifications@github.com>
napisał:

**Comment by Poincare** [comment]
@kamilkisiela Ok, this must be a bug then. I am now looking into it and will hopefully producing a test that reproduces it + a fix.

**Comment by Poincare** [comment]
@kamilkisiela So I tried to reproduce this issue within #456 but I couldn't quite do it. Could you look into the unit test - maybe I missing some crucial component that's actually causing the error?

**Comment by Poincare** [comment]
This must be related to #501 somehow.

**Comment by Poincare** [comment]
I can't reproduce this issue and it seems to have been fixed through #501 (going off your description).

**Comment by kamilkisiela** [comment]
@Poincare I've checked it with GitHunt for Angular2 and it seems to be fixed
